apps/test/unit/lib/kits/maker/boards/microBit/LightSensorTest.js

```diff
@@ -60,11 +69,62 @@ describe('LightSensor', function() {
       let stopSpy = sinon.spy(boardClient, 'stopStreamingAnalogChannel');
       lightSensor.start();
       expect(startSpy).to.have.been.calledOnce;
-      expect(startSpy).to.have.been.calledWith(sensor_channels.lightSensor);
+      expect(startSpy).to.have.been.calledWith(SENSOR_CHANNELS.lightSensor);
 
       lightSensor.stop();
       expect(stopSpy).to.have.been.calledOnce;
-      expect(stopSpy).to.have.been.calledWith(sensor_channels.lightSensor);
+      expect(stopSpy).to.have.been.calledWith(SENSOR_CHANNELS.lightSensor);
+    });
+  });
+
+  describe(`getAveragedValue`, () => {
+    it(`returns average of values in buffer`, () => {
+      lightSensor.state.buffer = new Float32Array(
+        MAX_SENSOR_BUFFER_DURATION / SAMPLE_INTERVAL
+      );
+      for (let i = 0; i < lightSensor.state.buffer.length; i++) {
+        lightSensor.state.buffer[i] = i;
+      }
+
+      // Test the average of the first second of data
+      lightSensor.state.currentBufferWriteIndex = 20;
+      expect(lightSensor.getAveragedValue(1000)).to.equal(9.5);
+
+      // Test that the full buffer can be averaged
+      lightSensor.state.currentBufferWriteIndex =
+        MAX_SENSOR_BUFFER_DURATION / SAMPLE_INTERVAL + 1;
+      expect(lightSensor.getAveragedValue(MAX_SENSOR_BUFFER_DURATION)).to.equal(
+        29.5
+      );
+
+      // Test that one index can be averaged
+      lightSensor.state.currentBufferWriteIndex = 1;
+      expect(lightSensor.getAveragedValue(SAMPLE_INTERVAL)).to.equal(0);
+
+      // Test the circular behavior of the buffer
+      lightSensor.state.currentBufferWriteIndex =
+        MAX_SENSOR_BUFFER_DURATION + 10;
```

Comment:
Should this be `MAX_SENSOR_BUFFER_DURATION / SAMPLE_INTERVAL + 10`?  It seems `MAX_SENSOR_BUFFER_DURATION` is on a different unit of measurement to the buffer itself...

Reply:
It shouldn't change the behavior since this behavior gets mod'ed, but I'll update for clarity.